### ISSUE TITLE
Send promise not rejected when disconnected

### ISSUE THREAD

**Issue by pierreca** [noodlefrenzy/node-amqp10#265]
Followup on https://github.com/Azure/azure-iot-sdks/issues/852

Repro steps: 
1. connect a client
2. create a sender
3. start sending messages at interval
4. verify it works
5. pull the network cable
6. wait

subsequent promises are never resolved or rejected, even after the disconnected event is received.
Please find below code/console output to repro:
```js
var amqp10 = require('amqp10');

var uri = 'amqps://sasl-uri';
var endpoint = '/endpoint';

var amqpClient = new amqp10.Client(amqp10.Policy.merge({
    senderLink: {
      attach: {
        properties: {
          'com.microsoft:client-version': 'amqp10test'
        },
        maxMessageSize: 0,
      },
      encoder: function(body) {
        if(typeof body === 'string') {
          return new Buffer(body, 'utf8');
        } else {
          return body;
        }
      },
      reattach: {
        retries: 0,
        forever: false
      }
    },
    receiverLink: {
      attach: {
        properties: {
          'com.microsoft:client-version': 'amqp10test'
        },
        maxMessageSize: 0,
        receiverSettleMode: false,
      },
      decoder: function(body) { return body; },
      reattach: {
        retries: 0,
        forever: false
      }
    },
    // reconnections will be handled at the client level, not the transport level.
    reconnect: {
      retries: 0,
      strategy: 'fibonnaci',
      forever: false
    }
  }, amqp10.Policy.EventHub));

amqpClient.on('connection:opened', function(ev) {
  console.log('connection:opened');
});

amqpClient.on('connection:closed', function(ev) {
  console.log('connection:closed');
});

amqpClient.on('client:errorReceived', function(err) {
  console.log('client:errorReceived: ' + err.toString());
});

amqpClient.connect(uri)
          .then(function() {
            return amqpClient.createSender(endpoint)
          })
          .then(startSending)
          .catch(function(err) {
            console.error(err.toString());
          });


function startSending(senderLink) {
  var msg = {
    properties: {
      to: endpoint
    },
    body: 'foo'
  };

  setInterval(function() {
    console.log('sending...')
    senderLink.send(msg)
            .then(function(res) {
              console.log('... sent: ');
            })
            .catch(function(err) {
              console.error(err.toString());
            });
  }, 1000);
}
```
```
connection:opened
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
... sent:
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
sending...
client:errorReceived: Error: read ECONNRESET
connection:closed
sending...
sending...
sending...
sending...
sending...
sending...
sending...
```

**Comment by 7chandan** [comment]
Hi, any idea on when this issue will be solved? Thanks.

**Comment by mbroadst** [comment]
@pierreca @7chandan I've pushed a fix to the `issue265` branch, please test it and confirm whether it fixes your problem.

**Comment by pierreca** [comment]
LGTM :shipit: :)

**Comment by mbroadst** [comment]
fixed in db93618f86415ff51d5f87226969400d4d4526f2
